support/ray_cluster_client.go

```diff
@@ -126,3 +188,33 @@ func (client *rayClusterClient) GetJobLogs(jobID string) (logs string, err error
 	err = json.Unmarshal(respData, &jobLogs)
 	return jobLogs.Logs, err
 }
+
+func (client *rayClusterClient) WaitForJobStatus(jobID string) (string, error) {
+	var status string
+	var prevStatus string
+	fmt.Printf("Waiting for job to be Succeeded...\n")
+	var err error
+	var resp *RayJobDetailsResponse
+	for status != "SUCCEEDED" {
```

Comment:
I'd suggest to have the polling logic done in the tests using Gomega which is more flexible than hard-coding the logic with a for loop.

Reply:
I have replaced for loop logic with the polling logic using gomega's eventually function.. Please review : https://github.com/project-codeflare/codeflare-common/pull/61/commits/e15c13214f544d0dcc8b6cac1541fd07d1b7cd1b

Thanks @astefanutti for above suggestion, it seems truely appropriate now !

Reply:
@abhijeet-dhumal The polling logic should be realized in test itself.
So in the codeflare-common you can return just `GetJobDetails`, in test you will use test.eventually to poll job status. You can place here a function which will extract job status from job details.

Reply:
@sutaakar @astefanutti  I have updated PR accordingly, Please review : 
https://github.com/project-codeflare/codeflare-common/pull/61/commits/b834fd92509583ec42d12a6cb3dd1dfcda086467
https://github.com/opendatahub-io/distributed-workloads/pull/208/files#diff-69184be2ae870ffa0507b3f31dfd4f8b00f3327306c386f395dacab0c0183635R152